android/demo-app/src/main/java/com/stadiamaps/ferrostar/DemoNavigationScene.kt

```diff
@@ -47,15 +44,10 @@ fun DemoNavigationScene(
     savedInstanceState: Bundle?,
     locationProvider: LocationProvider = AppModule.locationProvider,
 ) {
-  val executor = remember { Executors.newSingleThreadScheduledExecutor() }
-
   // Keeps the screen on at consistent brightness while this Composable is in the view hierarchy.
   KeepScreenOnDisposableEffect()
 
-  // NOTE: We are aware that this is not a particularly great pattern.
-  // We are working on improving this. See the discussion on
-  // https://github.com/stadiamaps/ferrostar/pull/295.
-  var viewModel by remember { mutableStateOf<NavigationViewModel>(DemoNavigationViewModel()) }
+  val viewModel = remember { DemoNavigationViewModel() }
```

Comment:
`DemoNavigationViewModel` can move to the AppModule to more accurately represent likely architecture.